### DISCUSSION TITLE
Extend theme

### DISCUSSION THREAD

**Discussion by IanDaly** [chakra-ui/chakra-ui#4600] [Q&A] [answered]
How can I extend the theme to include the additional colour schemes for the Input component ?

**Comment by chasinhues** [comment, marked as answer]
@IanDaly can you elaborate on what you mean by this?

**Reply by IanDaly** [reply]
Hi @chasinhues in the docs it says "Color Schemes for Input are not implemented in the default theme. You can extend the theme to implement them.", I am just not entirely sure how to extend the theme to implement the current available colour schemes to the input component. For example the purple colour scheme.

**Reply by chasinhues** [reply]
Ah yeah I get what you mean now, thanks for clarifying.

Check out this code sandbox for a demo of how to do this: https://codesandbox.io/s/add-colorscheme-support-to-input-theme-p12mn?file=/src/theme/components/Input.ts

### Highly recommended reading
1. How to customise the theme: https://chakra-ui.com/docs/theming/customize-theme
2. How to customise single-part and multi-part components: https://chakra-ui.com/docs/theming/component-style#single-part-and-multipart-components (especially, how to customise multi-part components)

### Important things to note
`Input` is a multi-part component. You can figure out if the component you're trying to customise is single-part or multi-part by going to the docs for that component and clicking the __View theme source__ button at the top of the page.

![image](https://user-images.githubusercontent.com/13565648/130387019-f028068b-845f-4fc8-8d0b-7bd72d127f5d.png)

**Reply by IanDaly** [reply]
Thank you, I appreciate the help
